src/redisearch_rs/buffer/src/lib.rs

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+//! The `Buffer` API Rust wrappers.
+//!
+//! This crate provides Rust wrappers for the `Buffer`, `BufferReader` and `BufferWriter` structs
+//! from `buffer.h`.
+
+use std::{
+    ptr::{NonNull, copy_nonoverlapping},
+    slice,
+};
+
+/// Redefines the `Buffer` struct from `buffer.h`
+#[repr(C)]
+pub struct Buffer {
+    /// A pointer to the underlying data buffer. This is typically allocated by C and hence should
+    /// not be freed by Rust code.
+    pub data: NonNull<u8>,
+    /// The capacity of the buffer (i-e allocated size)
+    pub capacity: usize,
+    /// The length of the buffer (i-e the total used memory from allocated memory)
+    pub len: usize,
```

Comment:
Should these fields be public? 
We have read-only accessors for capacity and length.
We also have `BufferWriter` to write into a buffer. 
There should be no need for public access outside this crate.